### ISSUE TITLE
C# InvalidProjectFileException - Invalid location

### ISSUE THREAD

**Issue by jpringle1** [dotnet/vscode-csharp#5953]
Type: <b>Bug</b>

## Issue Description ##
On startup, dotnet tries to open my .sln file, but always uses a jarbled path that consists of the intended project root path concatenated with the path to an .sln of one specific, unrelated project. For example:
`c:/repos/NewProject/c:/repos/ApiList/ApiList.sln`

The first half of the string changes depending on the project i'm trying to open, but the second half is always `c:/repos/workspaceRoot/project.sln`

## Steps to Reproduce ##
I just open vscode with C# extension enabled and it happens automatically
## Expected Behavior ##
Open {projectPath} instead of {projectPath}+{apiList.sln}
## Actual Behavior ##
Tries to open {projectPath}+{apiListSolutionPath}
### C# log ###
<details>
Dotnet path: c:\Users\joe.pringle\AppData\Roaming\Code\User\globalStorage\ms-dotnettools.vscode-dotnet-runtime\.dotnet\7.0.9\dotnet.exe

Activating C# standalone...

info: LanguageServerHost[0]
      Starting server...

[Info  - 9:31:14 AM] [LanguageServerProjectSystem]Loaded MSBuild at C:\Program Files\dotnet\sdk\7.0.102

[Info  - 9:31:14 AM] [LanguageServerProjectSystem]Loading c:\src\repos\jpringle\testProject\c:\src\repos\Assessment \ApiList\ApiList.sln...

[Error - 9:31:14 AM] [LanguageServerHost]Microsoft.Build.Exceptions.InvalidProjectFileException: The project file could not be loaded. 

The filename, directory name, or volume label syntax is incorrect. : 'c:\src\repos\jpringle\testProject\c:\src\repos\Assessment\ApiList\ApiList.sln'  c:\src\repos\jpringle\testProject\c:\src\repos\Assessment\ApiList\ApiList.sln
</details>

## Environment information ##

**VSCode version**: 1.80.1
**C# Extension**: 2.0.283
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>

Host:
  Version:      7.0.9
  Architecture: x64
  Commit:       8e9a17b221

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.NETCore.App 7.0.9 [c:\Users\joe.pringle\AppData\Roaming\Code\User\globalStorage\ms-dotnettools.vscode-dotnet-runtime\.dotnet\7.0.9\shared\Microsoft.NETCore.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|aspnetcorerazor-html-css-class-completion|kevin-chatham|1.0.3|kevin-chatham.aspnetcorerazor-html-css-class-completion-1.0.3|
|azure-repos|ms-vscode|0.36.0|ms-vscode.azure-repos-0.36.0|
|catppuccin-vsc|Catppuccin|2.7.0|catppuccin.catppuccin-vsc-2.7.0|
|codesnap|adpyke|1.3.4|adpyke.codesnap-1.3.4|
|csharp|ms-dotnettools|2.0.283|ms-dotnettools.csharp-2.0.283-win32-x64|
|data-workspace-vscode|ms-mssql|0.5.0|ms-mssql.data-workspace-vscode-0.5.0|
|deletecurrentfile|cha0ran|0.1.4|cha0ran.deletecurrentfile-0.1.4|
|dotnet-interactive-vscode|ms-dotnettools|1.0.4313020|ms-dotnettools.dotnet-interactive-vscode-1.0.4313020|
|gitlens|eamodio|14.1.1|eamodio.gitlens-14.1.1|
|html-related-links|rioj7|1.1.0|rioj7.html-related-links-1.1.0|
|jupyter|ms-toolsai|2023.6.1101941928|ms-toolsai.jupyter-2023.6.1101941928-win32-x64|
|jupyter-keymap|ms-toolsai|1.1.2|ms-toolsai.jupyter-keymap-1.1.2|
|jupyter-renderers|ms-toolsai|1.0.17|ms-toolsai.jupyter-renderers-1.0.17|
|live-server|ms-vscode|0.4.9|ms-vscode.live-server-0.4.9|
|material-icon-theme|PKief|4.29.0|pkief.material-icon-theme-4.29.0|
|mssql|ms-mssql|1.19.1|ms-mssql.mssql-1.19.1|
|namespace|adrianwilczynski|1.1.2|adrianwilczynski.namespace-1.1.2|
|path-intellisense|christian-kohler|2.8.4|christian-kohler.path-intellisense-2.8.4|
|powershell|ms-vscode|2023.6.0|ms-vscode.powershell-2023.6.0|
|remote-repositories|ms-vscode|0.38.1|ms-vscode.remote-repositories-0.38.1|
|rest-client|humao|0.25.1|humao.rest-client-0.25.1|
|sql-bindings-vscode|ms-mssql|0.4.0|ms-mssql.sql-bindings-vscode-0.4.0|
|sql-database-projects-vscode|ms-mssql|1.1.1|ms-mssql.sql-database-projects-vscode-1.1.1|
|test-adapter-converter|ms-vscode|0.1.8|ms-vscode.test-adapter-converter-0.1.8|
|tws|jkiviluoto|1.0.1|jkiviluoto.tws-1.0.1|
|vscode-dotnet-runtime|ms-dotnettools|1.6.0|ms-dotnettools.vscode-dotnet-runtime-1.6.0|
|vscode-edge-devtools|ms-edgedevtools|2.1.2|ms-edgedevtools.vscode-edge-devtools-2.1.2|
|vscode-extension-profiles|evald24|1.2.4|evald24.vscode-extension-profiles-1.2.4|
|vscode-gutter-preview|kisstkondoros|0.30.0|kisstkondoros.vscode-gutter-preview-0.30.0|
|vscode-jupyter-cell-tags|ms-toolsai|0.1.8|ms-toolsai.vscode-jupyter-cell-tags-0.1.8|
|vscode-jupyter-slideshow|ms-toolsai|0.1.5|ms-toolsai.vscode-jupyter-slideshow-0.1.5|
|vscode-nuget-gallery|patcx|0.0.24|patcx.vscode-nuget-gallery-0.0.24|
|vscode-toggle-quotes|BriteSnow|0.3.6|britesnow.vscode-toggle-quotes-0.3.6|;

</details>

Extension version: 2.0.283
VS Code version: Code 1.80.1 (74f6148eb9ea00507ec113ec51c489d6ffb4b771, 2023-07-12T17:22:07.651Z)
OS version: Windows_NT x64 10.0.19044
Modes:

<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30792226
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30610679
3biah626:30602489
89544117:30613380
a9j8j154:30646983
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
showindicator:30785052
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30779714
pythonidxpt:30784022
pythonnocebcf:30776496
e537b577:30795824
dsvsc013:30795093

```

</details>

**Comment by dibarbet** [comment]
Can you check what is in your `dotnet.defaultSolution` setting?  If you don't have a value for it, could you try setting it to your desired solution path and see if anything works?

**Comment by jasonmalinowski** [comment]
Also @jpringle1 was this a regular workspace or a multi-root workspace (where you have multiple folders all open at once)?

**Comment by jpringle1** [comment]
Annoyingly, this issue fixed itself today, and i can't think of anything that would have triggered it to fix itself since it's been a problem for weeks.

I will mark the issue as closed, but i suspect it may pop up again in the future.

**Comment by jasonmalinowski** [comment]
@jpringle1 If you see it again, feel free to reactivate. We might be able to psychic debug though if you have the answer to some of our questions though.

**Comment by jpringle1** [comment]
@jasonmalinowski It was originally a multi-root workspace, but I eventually converted it back to a single-root workspace in an attempt to fix it. The problem didn't go away for a number of days after this though.

As for the question from @dibarbet, I don't have a value for that setting in the user profile, but I have "ApiList.sln" in the project settings.

Interestingly, I tried changing the value to reproduce the bug, but found that vscode automatically rewrites it to the correct value every time i restart it. So maybe it's possible the default solution was incorrect in the user settings, but then vscode realised it was wrong and removed it.
